### PR TITLE
Adds SI unit formatting proc

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/52235
- https://github.com/tgstation/tgstation/pull/54600

## About The Pull Request
This adds an automatic formatting proc for SI conversions.

At the moment, its used only for mech charging and thermomachines, but in the future we can begin converting systems to use this.

Atomization of #8372 

## Why It's Good For The Game

Allows coders and mappers more flexibility, and better user experience with a non-fixed unit.

SI units are pretty important in physics, so it's kinda of a no-brainer to include them in an atmos-sim

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

testers

image

image

</details>

## Changelog
:cl: RKz, Donkie, MadMedicineMan
add: SI unit conversion proc and unit test, only used in thermomachines and mechbay chargers atm
/:cl:

### CODE REVIEW

**Review by itsmeow — CHANGES_REQUESTED** [review]
I would like this at least used somewhere

File: code/game/mecha/mech_bay.dm [hunk 1]
```diff
@@ -21,9 +22,12 @@
 	circuit = /obj/item/circuitboard/machine/mech_recharger
 	var/obj/mecha/recharging_mech
 	var/obj/machinery/computer/mech_bay_power_console/recharge_console
-	var/max_charge = 50
+	///Power unit per second to charge by, modified by /RefreshParts()
+	var/recharge_power = 25
```
Comment (qwertyquerty) [inline code/game/mecha/mech_bay.dm:26]: Why did this change

Comment (Tsar-Salat) [reply code/game/mecha/mech_bay.dm:26]: I added the reason in the comment. RefreshParts() overrides this default 50 with a 25.

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
I think that the procs that do stuff like kW, MW, kJ, MJ should be converted to siunit_power or whatever so we have the same system for all of them. Other than that, I don't think it necessarilly has to be used anywhere since I have had many cases where I wanted a proc like this

**Comment by Tsar-Salat** [comment]
> Other than that, I don't think it necessarilly has to be used anywhere since I have had many cases where I wanted a proc like this

So what do you want me to do here? Should I revert the commits that utilize the proc, and just leave it as an usused proc?

**Comment by Tsar-Salat** [comment]
i cant find anything else to implement else to implement it with

**Review by itsmeow — APPROVED** [review]

**Comment by itsmeow** [comment]
> i cant find anything else to implement else to implement it with.

Thermomachines, Supermatter, radiation collector
multitool on a wire

File: code/game/mecha/mech_bay.dm [hunk 2]
```diff
@@ -42,12 +46,12 @@
...
 	for(var/obj/item/stock_parts/capacitor/C in component_parts)
 		MC += C.rating
-	max_charge = MC * 25
+	recharge_power = MC * 25
 
 /obj/machinery/mech_bay_recharge_port/examine(mob/user)
 	. = ..()
 	if(in_range(user, src) || isobserver(user))
-		. += "<span class='notice'>The status display reads: Base recharge rate at <b>[max_charge]J</b> per cycle.</span>"
+		. += "<span class='notice'>The status display reads: Base recharge rate at <b>[siunit(recharge_power, "W", 1)]</b> per cycle.</span>"
```
Comment (PowerfulBacon) [inline code/game/mecha/mech_bay.dm]:
```suggestion
		. += "<span class='notice'>The status display reads: Base recharge rate at <b>[siunit(recharge_power, "W", 1)]</b>.</span>"
```

Comment (PowerfulBacon) [reply code/game/mecha/mech_bay.dm]: Otherwise the correct unit is Joules if we consider cycle the unit of time

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**